### PR TITLE
Replace token= keyword with name= in map_blocks

### PR DESCRIPTION
**Pull request by mrocklin**
Behavior here was reversed from normal dask.array conventions.
This does an abrupt change of the name= semantics and issues a warning
when users use token=, pointing them to name=

See https://github.com/dask/dask/issues/3226#issuecomment-396419864

- [ ] Tests added / passed
- [ ] Passes `flake8 dask`

### CODE REVIEW

**Comment by mrocklin** [comment]
cc @jakirkham

File: dask/array/core.py [hunk 1]
```diff
@@ -669,9 +669,11 @@ def map_blocks(func, *args, **kwargs):
         raise TypeError(msg % type(func).__name__)
     name = kwargs.pop('name', None)
     token = kwargs.pop('token', None)
-    if not name:
-        name = '%s-%s' % (token or funcname(func),
-                          tokenize(token or func, args, **kwargs))
+    if token:
+        warnings.warn("The token= keyword to map_blocks has been moved to name=")
```
Comment (jakirkham) [inline dask/array/core.py]: Should we note when we plan to drop `token`?

**Comment by jakirkham** [comment]
Thanks @mrocklin.

This seems fine to me. Had one question about the warning.

The CI failures seemed unrelated. So went ahead and restarted the failing builds.

File: dask/array/core.py (hunk 1 above)
Comment (mrocklin) [reply dask/array/core.py]: Perhaps?  When I run into warnings like this the future version number rarely means much to me.  I mostly care about "how far into the future will this work?"  the answer I get is some version number, which doesn't mean much to me, so I go ahead and change my code anyway.

Happy to adhere to some convention though if it exists.

Comment (mrocklin) [reply dask/array/core.py]: Any further concern here @jakirkham ?

Comment (jakirkham) [reply dask/array/core.py]: This point of including versions in deprecations came up once in a NumPy PR I submitted. So it seemed like a fair point to raise. That said, I don't have particularly strong feelings about it.

Comment (mrocklin) [reply dask/array/core.py]: OK, in that case I'll probably pass on this.

**Comment by mrocklin** [comment]
Thanks for the find @jakirkham .  I've gone ahead and remove the use of `name=` there altogether.  We should be able to rely on `map_blocks` to sniff out the function name in that case (this probably wasn't around when the original line was written).

**Pull request merged**